internal/oc/release.go

```diff
@@ -0,0 +1,136 @@
+package oc
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=release.go -package=oc -destination=mock_release.go
+type Release interface {
+	GetMCOImage(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error)
+	Extract(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, cacheDir string, pullSecret string) (string, error)
+}
+
+type release struct {
+}
+
+func NewRelease() Release {
+	return &release{}
+}
+
+func execute(log logrus.FieldLogger, pullSecret string, command string, args ...string) (string, error) {
+	// write pull secret to a temp file
+	ps, err := ioutil.TempFile("", "registry-config")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		ps.Close()
+		os.Remove(ps.Name())
+	}()
+	_, err = ps.Write([]byte(pullSecret))
+	if err != nil {
+		return "", err
+	}
+	// flush the buffer to ensure the file can be read
+	ps.Close()
+
+	args = append(args, "--registry-config="+ps.Name())
+
+	cmd := exec.Command(command, args...)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &out
+	err = cmd.Run()
+	if err != nil {
+		log.Error(out.String())
+		return "", err
+	}
+	return strings.TrimSpace(out.String()), nil
+}
+
+// GetMCOImage gets mcoImage url from the releaseImageMirror if provided.
+// Else gets it from the source releaseImage
+func (r *release) GetMCOImage(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error) {
+	var mcoImage string
+	var err error
+	if releaseImageMirror != "" {
+		//TODO: Get mirror registry certificate from install-config
+		mcoImage, err = getMCOImageFromRelease(log, releaseImageMirror, pullSecret, true)
+		if err != nil {
+			log.Errorf("failed to get mco image from mirror release image %s", releaseImageMirror)
```

Comment:
lets have it with log with error. It will be much easier to debug

Reply:
Done